pkg/operator/mds/mds.go

```diff
@@ -0,0 +1,207 @@
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package mds
+
+import (
+	"fmt"
+
+	"github.com/rook/rook/pkg/cephmgr/client"
+	cephmds "github.com/rook/rook/pkg/cephmgr/mds"
+	"github.com/rook/rook/pkg/cephmgr/mon"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	k8smon "github.com/rook/rook/pkg/operator/mon"
+	"k8s.io/client-go/1.5/kubernetes"
+	"k8s.io/client-go/1.5/pkg/api/v1"
+	extensions "k8s.io/client-go/1.5/pkg/apis/extensions/v1beta1"
+)
+
+const (
+	appName            = "mds"
+	dataPoolSuffix     = "-data"
+	metadataPoolSuffix = "-metadata"
+	keyringName        = "keyring"
+)
+
+type Cluster struct {
+	Namespace string
+	Version   string
+	Replicas  int32
+	factory   client.ConnectionFactory
+}
+
+func New(namespace, version string, factory client.ConnectionFactory) *Cluster {
+	return &Cluster{
+		Namespace: namespace,
+		Version:   version,
+		Replicas:  1,
+		factory:   factory,
+	}
+}
+
+func (c *Cluster) Start(clientset *kubernetes.Clientset, cluster *mon.ClusterInfo) error {
+	logger.Infof("start running mds")
+
+	if cluster == nil || len(cluster.Monitors) == 0 {
+		return fmt.Errorf("missing mons to start mds")
+	}
+
+	context := &clusterd.DaemonContext{ConfigDir: "/var/lib/rook"}
+	conn, err := mon.ConnectToClusterAsAdmin(clusterd.ToContext(context), c.factory, cluster)
+	if err != nil {
+		return fmt.Errorf("failed to connect to cluster as admin: %+v", err)
+	}
+	defer conn.Shutdown()
+
+	// creating a named file system will be done with third party resources in the future
+	err = c.createFilesystem("myfs", "myfspool", cluster, conn)
+	if err != nil {
+		return fmt.Errorf("failed to create file system. %+v", err)
+	}
+
+	id := "mds1"
+	err = c.createKeyring(clientset, context, cluster, conn, id)
+	if err != nil {
+		return fmt.Errorf("failed to create mds keyring. %+v", err)
+	}
+
+	// start the deployment
+	deployment, err := c.makeDeployment(cluster, id)
+	_, err = clientset.Deployments(c.Namespace).Create(deployment)
+	if err != nil {
+		if !k8sutil.IsKubernetesResourceAlreadyExistError(err) {
+			return fmt.Errorf("failed to create mds deployment. %+v", err)
+		}
+		logger.Infof("mds deployment already exists")
+	} else {
+		logger.Infof("mds deployment started")
+	}
+
+	return nil
+}
+
+func (c *Cluster) createFilesystem(id, poolName string, cluster *mon.ClusterInfo, conn client.Connection) error {
+	_, err := client.GetFilesystem(conn, id)
+	if err == nil {
+		logger.Infof("file system %s already exists", id)
+	}
+	logger.Warningf("Check for existence and failed to get file system %s. %+v", id, err)
```

Comment:
Oh, i meant to remove this warning. If the operator starts up again, it will skip creating the file system since it already exists.